homeassistant/components/asuswrt/config_flow.py

```diff
@@ -80,25 +80,28 @@ def _show_setup_form(
         if user_input is None:
             user_input = {}
 
+        adv_schema = {}
+        conf_password = vol.Required(CONF_PASSWORD)
+        if self.show_advanced_options:
+            conf_password = vol.Optional(CONF_PASSWORD)
+            adv_schema = {
```

Comment:
We don't need to redefine the dict we can just set the items in it.

Reply:
Done.👍